qubes/tests/integ/basic.py

```diff
@@ -623,8 +623,12 @@ def test_010_os_metadata(self):
             self.assertIsNotNone(tpl.features.get('os-eol'))
         elif self.template.startswith('whonix-'):
             self.assertEqual(tpl.features.get('os-distribution'), 'whonix')
+            self.assertEqual(tpl.features.get('os-distribution-like'), 'debian')
```

Comment:
This test failed for (both) Whonix templates, the bot will post details when the run completes. I do see in the log it sent some info to dom0 after installing updates:
```
Jun 10 08:27:55 dom0 qrexec-policy-daemon[19866]: qrexec: qubes.FeaturesRequest+: whonix-workstation-17 -> @adminvm: allowed to dom0
```

Reply:
This depends on the other PR https://github.com/QubesOS/qubes-core-agent-linux/pull/504

```sh
    elif [ -f /usr/share/whonix/marker ]; then
        distro="whonix"
        distro_like="debian"
```

Because Whonix does not set `/etc/os-release` to the norm of using `ID_LIKE=`, the value will just be reported correctly after the other PR is merged. I my assumption correct?

Reply:
I also assume that Whonix reposted an empty value for `os-distribution-like` as Whonix does not set `ID_LIKE=`. It will just work after the other PR is merged unless tests could be made upon the other PR branch also instead of the main branch.

Reply:
The test did run with the other PR included too...

Reply:
Ok, I do not know the issue yet. Last commit only fixes the version gathering for Whonix and Kicksecure which where inverted.

Reply:
Ok, the issue was with the version gathering, I inverted Whonix and Kicksecure information and with `set -e`, it was exiting on the first error:

```sh
+ [ -f /usr/share/kicksecure/marker ]
+ [ -f /usr/share/whonix/marker ]
+ distro=whonix
+ distro_like=debian
+ cat /etc/kicksecure_version
cat: /etc/kicksecure_version: No such file or directory
+ version=
zsh: exit 1     sudo sh -x ./10-qubes-core-agent-features.sh
```

Last commit https://github.com/QubesOS/qubes-core-agent-linux/commit/c33b577544bf92a328545a183e77f344047935f2 fixes this issue. [Comparison](https://github.com/QubesOS/qubes-core-agent-linux/compare/0e9290134165f3ef663eaaa48102fc3f77c81cc5..c33b577544bf92a328545a183e77f344047935f2).